### DISCUSSION TITLE
cannot select model type ---- Please Help

### DISCUSSION THREAD

**Discussion by braamrichardchristopher** [laravel/framework#33378] [General]
how would i get the type name of my referenced table in my get

Here is my database example:

    mainTable
    id | typeid
    1  | 1
    2  | 1
    3  | 2

And here is my second table

    typesTable
    id | type
    1  | test
    2  | test2
    3  | test3

My relationship in my main table model is defined as :

    public function type()
    {
        return $this->belongsTo('App\Models\typesTable', 'typeID');
    }

And here is how im pulling the data:

    try {
                $data = Main::with('type')
                ->groupBy('typeID')
                ->get(array(
                    DB::raw('type as date') <---getting type column does not exist-->,
                    DB::raw('COUNT(*) as "views"')
                ));
    
                return Respond::Ok('Success', $data);
            } catch (\Throwable $th) {
                return Respond::Bad('Failed', $th);
            }
my problem is i cannot get the type name in my get array and am getting an error..

if i remove that line i get the following data back:

    message: "Success"
    data: 
    0: {views: 2, type: null}
    1: {views: 1, type: null}
    2: {views: 1, type: null}
    meta: null

Please help

**Comment by rcbgalido** [comment]
May need to update `typeID` to `typeid`.
```
public function type()
{
    return $this->belongsTo('App\Models\typesTable', 'typeid');
}
```
